dockerpty/io.py

```diff
@@ -210,6 +251,33 @@ def write(self, data):
 
         return self.stream.write(data)
 
+    def needs_write(self):
+        """
+        Delegates to underlying Stream.
+        """
+
+        if hasattr(self.stream, 'needs_write'):
+            return self.stream.needs_write()
```

Comment:
Do we expect streams that don't have these attribute? From what I can see this is only used with `io.Stream`